completions/bind

```diff
@@ -1,16 +1,73 @@
 # bash bind completion                                     -*- shell-script -*-
 
+_bind_var_vals()
+{
+    local pre=$1 var=$2 val=$3 vals
+    case $var in
+        bind-tty-special-chars|blink-matching-paren|byte-oriented|\
+        colored-completion-prefix|colored-stats|completion-ignore-case|\
+        completion-map-case|convert-meta|disable-completion|\
+        echo-control-characters|enable-bracketed-paste|enable-keypad|\
+        enable-meta-key|expand-tilde|history-preserve-point|\
+        horizontal-scroll-mode|input-meta|mark-directories|\
+        mark-modified-lines|mark-symlinked-directories|match-hidden-files|\
+        menu-complete-display-prefix|meta-flag|output-meta|page-completions|\
+        prefer-visible-bell|print-completions-horizontally|\
+        revert-all-at-newline|show-all-if-ambiguous|show-all-if-unmodified|\
+        show-mode-in-prompt|skip-completed-text|visible-stats)
+            vals=(on off)
+            ;;
+        bell-style)
+            vals=(audible none visible)
+            ;;
+        editing-mode)
+            vals=(emacs vi)
+            ;;
+        keymap)
+            vals=(emacs emacs-standard emacs-meta emacs-ctlx vi vi-command
+                vi-insert)
+            ;;
+        *)
+            return 1
+            ;;
+    esac
+    local v reset=$(shopt -p nocasematch); shopt -s nocasematch
+    for v in "${vals[@]}"; do
+        [[ $v == "$val"* ]] && COMPREPLY+=("${pre}${val}${v/#$val}")
+    done
+    eval "$reset"
```

Comment:
I think a plain `$reset` should work here, no need to eval. And the `shopt` probably needs protection from nondefault `$IFS`. See existing similar constructs, commit 5240618033fecc3032a421d39c2eaeb2e4c20d57 and #143

Reply:
I was thinking `eval "$reset"` might be a better/cleaner construct to use in these cases, rather than doing the `$IFS` switcheroo, since `eval "$reset"` works the same way regardless of `$IFS` value and so doesn't require protection from non-default values.  Curious what you think.

Reply:
Consistency and eval avoidance are good things to pursue.